### PR TITLE
[IMP] web_responsive: Add search feature

### PR DESCRIPTION
**Pull request by yajo**
Indludes a squashed and slightly altered version of @lasley's #838, and supersedes that stale PR.

You should be able to search menus now.

@Tecnativa

### CODE REVIEW

File: web_responsive/readme/DESCRIPTION.rst [hunk 1]
```diff
@@ -2,7 +2,22 @@ This module provides a mobile compliant interface for Odoo Community web.
...
- * Keyboard shortcuts for easier navigation
- * Display kanban views for small screens if an action or field One2x
- * Set chatter side (Optional per user)
+* New navigation with an App drawer
+* Keyboard shortcuts for easier navigation
+* Display kanban views for small screens if an action or field One2x
+* Set chatter side (Optional per user)
+* Quick search (see below)
+
+The following keyboard shortcuts are implemented:
```
Comment (pedrobaeza) [inline web_responsive/readme/DESCRIPTION.rst]: This should be in "Usage" section, isn't it?

**Review by pedrobaeza — DISMISSED** [review]
Travis is red, and #850 is also present here. There should a method that returns the available menus instead of searching the ir.ui.menu entries.

**Comment by pedrobaeza** [comment]
Please solve the conflict now that I have merged the other PR, and can you try to find a cleanest way of solving the menu issue. Standard menu and enterprise doesn't need these methods.

**Comment by pedrobaeza** [comment]
@Yajo what is the status of this?

**Comment by yajo** [comment]
I rebased, I wasn't aware of the git conflicts.

Current implementation is worse than enterprise because, as you said, it should use the cached menus instead of asking the server, but it's much better than it was in v10 because it defers the search until user stops writing and it aborts current search if he starts writing again, and because it fixes #850.

Implementing enterprise-like search would probably mean refactoring the addon, which is actually quite messy.

All of this is written in the known issues section. I wouldn't personally block enhancements just for not being perfect, mostly when they are much better than what was there before, but I'm not the one to decide on that. If you prefer to block it until it's perfect, just be aware it will take some time.

Besides all of that, it's still missing reviews.

**Comment by pedrobaeza** [comment]
Using this PR, half of the menus disappear not in the search, but on the top part. Can you please check?

**Comment by yajo** [comment]
I cannot reproduce :confused:

**Comment by pedrobaeza** [comment]
Put it in our instance as a test and you'll see the problem.

**Comment by Tardo** [comment]
Some error on runbot:
error_odoo

Login as "demo" user cause mistakes...
- When no app available in the menu, if use search feature and press enter you get an "undefined" url.
- When app available you can't navigate... do a click on icon doesn't launch any action.

**Comment by yajo** [comment]
> Some error on runbot:

This comes from the `web_action_conditionable` addon, which asking for a dependency that doesn't exist (`web.View`). It's unrelated AFAICS.



> When app available you can't navigate... do a click on icon doesn't launch any action.

It seems to be a cache problem. Reloading without cache worked for me. Can you check please?

> When no app available in the menu, if use search feature and press enter you get an "undefined" url.

It should be fixed now.

**Comment by yajo** [comment]
> half of the menus disappear not in the search, but on the top part

I finally managed to reproduce this, I'll work on the fix. Thanks!

**Comment by yajo** [comment]
It should work now.

**Review by tarteo — CHANGES_REQUESTED** [review]
Cool feature!

Is it possible to search on the display_name instead of the name e.g. 'Settings' should also find menu 'Settings/Dashboard'.

**Comment by yajo** [comment]
It's not possible because it is a computed, non-stored, context-dependant field.

The only way would be doing an offline search, purely in JS, through the cached visible menus, as `web_enterprise` does, but that'd mean a refactor of this addon, which is needed anyway BTW, but honestly I'd try to tackle that when migrating to v12.

**Review by tarteo — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]
Please squash commits before merging

**Comment by pedrobaeza** [comment]
I have squashed already, so I merge.

**Pull request merged**